File: BPMOPMobile_VietNamAirLines/BPMOPMobile_VietNamAirLines/BPMOPMobile.Droid/Presenter/SharedView/SharedView_PopupChooseFile.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using Android;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Provider;
using Android.Runtime;
using Android.Support.V4.App;
using Android.Support.V4.Content;
using Android.Support.V7.App;
using Android.Support.V7.Widget;
using Android.Util;
using Android.Views;
using Android.Widget;
using BPMOPMobile.Class;
using BPMOPMobile.Droid.Class;
using BPMOPMobile.Droid.Core.Common;
using BPMOPMobile.Droid.Core.Controller;
using BPMOPMobile.Droid.Presenter.Fragment;

namespace BPMOPMobile.Droid.Presenter.SharedView
{
    public class SharedView_PopupChooseFile : SharedViewBase
    {
        public int _requestCodeFile;
        public int _requestCodeCamera;
        public int _flagview;

        public enum FlagView
        {
            [Description("DetailWorkflow - ControlInputAttachmentVertical")]
            DetailWorkflow_ControlInputAttachmentVertical = 0,

            [Description("DetailWorkflow - Comment")]
            DetailWorkflow_Comment = 1,

            [Description("DetailCreateTask - ControlInputAttachmentVertical")]
            DetailCreateTask_ControlInputAttachmentVertical = 2,

            [Description("DetailCreateTask - Comment")]
            DetailCreateTask_Comment = 3,

            [Description("DetailCreateTask_Child - ControlInputAttachmentVertical")]
            DetailCreateTask_Child_ControlInputAttachmentVertical = 4,

            [Description("ReplyComment")]
            ReplyComment = 5,
        }

        public SharedView_PopupChooseFile(LayoutInflater _inflater, AppCompatActivity _mainAct, CustomBaseFragment _fragment, string _fragmentTag, View _rootView,
            int _requestCodeFile, int _requestCodeCamera, int _flagview) : base(_inflater, _mainAct, _fragment, _fragmentTag, _rootView)
        {
            this._requestCodeFile = _requestCodeFile;
            this._requestCodeCamera = _requestCodeCamera;
            this._flagview = _flagview;
        }

        public override void InitializeView()
        {
            ControllerBase CTRLBase = new ControllerBase();
            try
            {
                #region Get View - Init Data
                DisplayMetrics _displayMetrics = _fragment.Resources.DisplayMetrics;
                View _viewPopupControl = _inflater.Inflate(Resource.Layout.PopupControlAttachmentChooseFile, null);

                ImageView _imgBack = _viewPopupControl.FindViewById<ImageView>(Resource.Id.img_PopupControlAttachmentChooseFile_Back);
                TextView _tvTitle = _viewPopupControl.FindViewById<TextView>(Resource.Id.tv_PopupControlAttachmentChooseFile_title);
                TextView _tvInApp = _viewPopupControl.FindViewById<TextView>(Resource.Id.tv_PopupControlAttachmentChooseFile_InApp);
                RecyclerView _recyInApp = _viewPopupControl.FindViewById<RecyclerView>(Resource.Id.recy_PopupControlAttachmentChooseFile_InApp);
                ListView _lvInApp = _viewPopupControl.FindViewById<ListView>(Resource.Id.lv_PopupControlAttachmentChooseFile_InApp);
                TextView _tvOther = _viewPopupControl.FindViewById<TextView>(Resource.Id.tv_PopupControlAttachmentChooseFile_Other);
                LinearLayout _lnOtherCloud = _viewPopupControl.FindViewById<LinearLayout>(Resource.Id.ln_PopupControlAttachmentChooseFile_Other_Cloud);
                LinearLayout _lnOtherLibrary = _viewPopupControl.FindViewById<LinearLayout>(Resource.Id.ln_PopupControlAttachmentChooseFile_Other_Library);
                LinearLayout _lnOtherCamera = _viewPopupControl.FindViewById<LinearLayout>(Resource.Id.ln_PopupControlAttachmentChooseFile_Other_Camera);
                TextView _tvOtherCloud = _viewPopupControl.FindViewById<TextView>(Resource.Id.tv_PopupControlAttachmentChooseFile_Other_Cloud);
                TextView _tvOtherLibrary = _viewPopupControl.FindViewById<TextView>(Resource.Id.tv_PopupControlAttachmentChooseFile_Other_Library);
                TextView _tvOtherCamera = _viewPopupControl.FindViewById<TextView>(Resource.Id.tv_PopupControlAttachmentChooseFile_Other_Camera);
                CTRLBase.RequestAppPermission(_mainAct);
                if (CmmVariable.SysConfig.LangCode.Equals(CmmDroidVariable.M_SysLangVN))
                {
                    _tvTitle.Text = CmmFunction.GetTitle("TEXT_ATTACHMENT", "Tài liệu đính kèm");
                    _tvInApp.Text = CmmFunction.GetTitle("TEXT_FILE_INAPP", "Tập tin trong ứng dụng");
                    _tvOther.Text = CmmFunction.GetTitle("TEXT_OTHER_RESOURCE", "Nguồn khác");
                    _tvOtherCloud.Text = CmmFunction.GetTitle("TEXT_STORAGE_APPLICATION", "Ứng dụng lưu trữ");
                    _tvOtherLibrary.Text = CmmFunction.GetTitle("TEXT_PHOTO_LIBRARY", "Thư viện ảnh");
                    _tvOtherCamera.Text = "Camera";
                }
                else
                {
                    _tvTitle.Text = CmmFunction.GetTitle("TEXT_ATTACHMENT", "Attached file");
                    _tvInApp.Text = CmmFunction.GetTitle("TEXT_FILE_INAPP", "Files in app");
                    _tvOther.Text = CmmFunction.GetTitle("TEXT_OTHER_RESOURCE", "Other resources");
                    _tvOtherCloud.Text = CmmFunction.GetTitle("TEXT_STORAGE_APPLICATION", "Storage application");
                    _tvOtherLibrary.Text = CmmFunction.GetTitle("TEXT_PHOTO_LIBRARY", "Photo library");
                    _tvOtherCamera.Text = "Camera";
                }

                #endregion

                #region Show View
                Dialog _dialogAction = new Dialog(_rootView.Context, Resource.Style.Theme_Custom_BPMOP_Dialog_FullScreen_Animation);
                Window window = _dialogAction.Window;
                var dm = _fragment.Resources.DisplayMetrics;
                window.SetSoftInputMode(SoftInput.AdjustPan); // Để Hide soft keyboard ko bị Resize
                window.SetGravity(GravityFlags.Center);

                _dialogAction.RequestWindowFeature(1);
                _dialogAction.SetCanceledOnTouchOutside(false);
                _dialogAction.SetCancelable(true);
                _dialogAction.SetContentView(_viewPopupControl);
                _dialogAction.Show();

                WindowManagerLayoutParams s = window.Attributes;
                s.Width = WindowManagerLayoutParams.MatchParent;
                s.Height = WindowManagerLayoutParams.MatchParent;
                window.Attributes = s;
                #endregion

                #region Event

                _imgBack.Click += (sender, e) =>
                {
                    _dialogAction.Dismiss();
                };
                _lnOtherCloud.Click += (sender, e) =>
                {
                    try
                    {
                        Intent intent = new Intent(Intent.ActionGetContent);
                        intent.PutExtra(Intent.ExtraMimeTypes, CmmDroidVariable.M_MimeTypes);
                        intent.SetFlags(ActivityFlags.ClearTop);
                        if (Build.VERSION.SdkInt < BuildVersionCodes.O)
                            intent.SetType("*/*");
                        else
                            intent.SetType("file/*");
                        _fragment.StartActivityForResult(intent, _requestCodeFile);

                        _dialogAction.Dismiss();

                    }
                    catch (Exception) { }
                };
                _lnOtherLibrary.Click += (sender, e) =>
                {
                    try
                    {
                        Intent intent = new Intent(Intent.ActionGetContent);
                        intent.PutExtra(Intent.ExtraMimeTypes, CmmDroidVariable.M_MimeTypesImage);
                        intent.SetFlags(ActivityFlags.ClearTop);
                        if (Build.VERSION.SdkInt < BuildVersionCodes.O)
                            intent.SetType("*/*");
                        else
                            intent.SetType("file/*");
                        _fragment.StartActivityForResult(intent, _requestCodeFile);

                        _dialogAction.Dismiss();
                    }
                    catch (Exception) {
                        _mainAct.RunOnUiThread(() =>
                        {
                            Intent intent = new Intent(Settings.ActionApplicationDetailsSettings);
                            Android.Net.Uri uri = Android.Net.Uri.FromParts("package", _mainAct.PackageName, null);
                            intent.SetData(uri);
                            _mainAct.StartActivity(intent);
                        });
                    }
                };
                _lnOtherCamera.Click += (sender, e) =>
                {
                    try
                    {
                        /*if (ContextCompat.CheckSelfPermission(_mainAct, Manifest.Permission.Camera) != Android.Content.PM.Permission.Granted ||
                        ContextCompat.CheckSelfPermission(_mainAct, Manifest.Permission.WriteExternalStorage) != Android.Content.PM.Permission.Granted ||
                        ContextCompat.CheckSelfPermission(_mainAct, Manifest.Permission.ReadExternalStorage) != Android.Content.PM.Permission.Granted)
                        {
                            ActivityCompat.RequestPermissions(_mainAct,
                                        new string[] { Manifest.Permission.Camera, Manifest.Permission.WriteExternalStorage, Manifest.Permission.ReadExternalStorage }, CmmDroidVariable.M_DetailWorkflow_ChooseFileComment_Camera);
                        }
                        else
                        {*/
                        string filePath = System.IO.Path.Combine(CmmVariable.M_Folder_Avatar + "/", CmmDroidVariable.M_Camera_Prefix + SystemClock.ElapsedRealtime() + ".jpg");
                        Java.IO.File fileFromCamera = null;


                        switch (_flagview)
                        {
                            case (int)FlagView.DetailWorkflow_ControlInputAttachmentVertical:
                            case (int)FlagView.DetailWorkflow_Comment:
                                {
                                    FragmentDetailWorkflow _currentFragment = (FragmentDetailWorkflow)_fragment;
                                    _currentFragment._tempfileFromCamera = new Java.IO.File(filePath);
                                    fileFromCamera = _currentFragment._tempfileFromCamera;
                                    break;
                                }
                            case (int)FlagView.DetailCreateTask_ControlInputAttachmentVertical:
                            case (int)FlagView.DetailCreateTask_Comment:
                                {
                                    FragmentDetailCreateTask _currentFragment = (FragmentDetailCreateTask)_fragment;
                                    _currentFragment._tempfileFromCamera = new Java.IO.File(filePath);
                                    fileFromCamera = _currentFragment._tempfileFromCamera;
                                    break;
                                }
                            case (int)FlagView.DetailCreateTask_Child_ControlInputAttachmentVertical:
                                {
                                    FragmentDetailCreateTask_Child _currentFragment = (FragmentDetailCreateTask_Child)_fragment;
                                    _currentFragment._tempfileFromCamera = new Java.IO.File(filePath);
                                    fileFromCamera = _currentFragment._tempfileFromCamera;
                                    break;
                                }
                            case (int)FlagView.ReplyComment:
                                {
                                    FragmentReplyComment _currentFragment = (FragmentReplyComment)_fragment;
                                    _currentFragment._tempfileFromCamera = new Java.IO.File(filePath);
                                    fileFromCamera = _currentFragment._tempfileFromCamera;
                                    break;
                                }
                        }

                        if (fileFromCamera != null)
                        {
                            try
                            {
                                fileFromCamera.CreateNewFile();
                            }
                            catch (System.IO.IOException) { }
                            Android.Net.Uri outputURI = Android.Net.Uri.FromFile(fileFromCamera);

                            Intent intent = new Intent(MediaStore.ActionImageCapture);
                            intent.PutExtra(MediaStore.ExtraOutput, FileProvider.GetUriForFile(_mainAct, CmmDroidVariable.M_PackageProvider, fileFromCamera));
                            _fragment.StartActivityForResult(intent, _requestCodeCamera);
                        }
                        _dialogAction.Dismiss();
                    }
                    //}
                    catch (Exception ex)
                    {
                        _mainAct.RunOnUiThread(() =>
                        {
                            Intent intent = new Intent(Settings.ActionApplicationDetailsSettings);
                            Android.Net.Uri uri = Android.Net.Uri.FromParts("package", _mainAct.PackageName, null);
                            intent.SetData(uri);
                            _mainAct.StartActivity(intent);
                        });
                    }
                };

                #endregion
            }
            catch (Exception ex)
            {
#if DEBUG
                CmmDroidFunction.WriteTrackingError(this.GetType().Name, "InitializeView", ex);
#endif
            }
        }
    }
}

File: VNABPMOP/app/src/main/java/com/vuthao/bpmop/shareview/SharedView_PopupChooseFile.java
package com.vuthao.bpmop.shareview;

import android.Manifest;
import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.Dialog;
import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.provider.MediaStore;
import android.util.Log;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.activity.result.ActivityResultLauncher;
import androidx.core.content.FileProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.gson.Gson;
import com.vuthao.bpmop.R;
import com.vuthao.bpmop.base.BroadcastUtility;
import com.vuthao.bpmop.base.Constants;
import com.vuthao.bpmop.base.Functions;
import com.vuthao.bpmop.base.activity.BaseActivity;
import com.vuthao.bpmop.base.activity.BaseFragment;
import com.vuthao.bpmop.base.model.app.CurrentUser;
import com.vuthao.bpmop.base.model.custom.DownloadedFiles;
import com.vuthao.bpmop.base.realm.RealmController;
import com.vuthao.bpmop.base.vars.VarsReceiver;
import com.vuthao.bpmop.core.Vars;
import com.vuthao.bpmop.detail.DetailWorkflowActivity;
import com.vuthao.bpmop.detail.fragment.ReplyCommentFragment;
import com.vuthao.bpmop.shareview.adapter.DownloadedFilesAdapter;
import com.vuthao.bpmop.task.DetailCreateTaskActivity;
import com.vuthao.bpmop.task.child.CreateChildTaskActivity;

import java.io.File;
import java.io.IOException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;

import io.realm.RealmResults;

public class SharedView_PopupChooseFile extends SharedViewBase implements View.OnClickListener, DownloadedFilesAdapter.DownloadedFilesListener {
    public int requestCodeFile;
    public int requestCodeCamera;
    public int flagview;
    private ActivityResultLauncher<Intent> mLauncherNormal;
    private ActivityResultLauncher<Intent> mLauncherCamera;
    private BaseFragment baseFragment;

    private Dialog dialog;
    private View view;
    private ImageView imgBack;
    private TextView tvTitle;
    private TextView tvInApp;
    private TextView tvOther;
    private LinearLayout lnOtherCloud;
    private LinearLayout lnOtherLibrary;
    private LinearLayout lnOtherCamera;
    private TextView tvOtherCloud;
    private TextView tvOtherLibrary;
    private TextView tvOtherCamera;
    private TextView tvNodata;
    private RecyclerView lvInApp;

    public SharedView_PopupChooseFile(LayoutInflater inflater, Activity mainAct, String fragmentTag, View rootView,
                                      int requestCodeFile, int requestCodeCamera, int flagview) {
        super(inflater, mainAct, fragmentTag, rootView);
        this.requestCodeFile = requestCodeFile;
        this.requestCodeCamera = requestCodeCamera;
        this.flagview = flagview;
    }

    // Dành cho ReplyComment
    public SharedView_PopupChooseFile(LayoutInflater inflater, Activity mainAct, BaseFragment baseFragment, ActivityResultLauncher<Intent> mLauncherNormal, ActivityResultLauncher<Intent> mLauncherCamera, String fragmentTag, View rootView,
                                      int requestCodeFile, int requestCodeCamera, int flagview) {
        super(inflater, mainAct, fragmentTag, rootView);
        this.requestCodeFile = requestCodeFile;
        this.requestCodeCamera = requestCodeCamera;
        this.flagview = flagview;
        this.mLauncherNormal = mLauncherNormal;
        this.mLauncherCamera = mLauncherCamera;
        this.baseFragment = baseFragment;
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void initializeView() {
        view = inflater.inflate(R.layout.popup_control_attachment_choose_file, null);
        imgBack = view.findViewById(R.id.img_PopupControlAttachmentChooseFile_Back);
        tvTitle = view.findViewById(R.id.tv_PopupControlAttachmentChooseFile_title);
        tvInApp = view.findViewById(R.id.tv_PopupControlAttachmentChooseFile_InApp);
        tvOther = view.findViewById(R.id.tv_PopupControlAttachmentChooseFile_Other);
        lnOtherCloud = view.findViewById(R.id.ln_PopupControlAttachmentChooseFile_Other_Cloud);
        lnOtherLibrary = view.findViewById(R.id.ln_PopupControlAttachmentChooseFile_Other_Library);
        lnOtherCamera = view.findViewById(R.id.ln_PopupControlAttachmentChooseFile_Other_Camera);
        tvOtherCloud = view.findViewById(R.id.tv_PopupControlAttachmentChooseFile_Other_Cloud);
        tvOtherLibrary = view.findViewById(R.id.tv_PopupControlAttachmentChooseFile_Other_Library);
        tvOtherCamera = view.findViewById(R.id.tv_PopupControlAttachmentChooseFile_Other_Camera);
        tvNodata = view.findViewById(R.id.tv_PopupControlAttachmentChooseFile_NoData);
        lvInApp = view.findViewById(R.id.recy_PopupControlAttachmentChooseFile_InApp);

        tvTitle.setText(Functions.share.getTitle("TEXT_ATTACHMENT", "Tài liệu đính kèm"));
        tvNodata.setText(Functions.share.getTitle("TEXT_NODATA", "No data"));
        tvInApp.setText(Functions.share.getTitle("TEXT_FILE_INAPP", "Tập tin trong ứng dụng"));
        tvOther.setText(Functions.share.getTitle("TEXT_OTHER_RESOURCE", "Nguồn khác"));
        tvOtherCloud.setText(Functions.share.getTitle("TEXT_STORAGE_APPLICATION", "Ứng dụng lưu trữ"));
        tvOtherLibrary.setText(Functions.share.getTitle("TEXT_PHOTO_LIBRARY", "Thư viện ảnh"));
        tvOtherCamera.setText("Camera");

        loadDownloadedFiles();

        // region Dialog
        dialog = new Dialog(mainAct, R.style.Theme_Custom_BPMOP_Dialog_FullScreen_Animation);
        Window window = dialog.getWindow();
        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
        window.setGravity(Gravity.CENTER);
        dialog.requestWindowFeature(1);
        dialog.setCanceledOnTouchOutside(false);
        dialog.setCancelable(true);
        dialog.setContentView(view);
        dialog.show();

        WindowManager.LayoutParams s = window.getAttributes();
        s.width = WindowManager.LayoutParams.MATCH_PARENT;
        s.height = WindowManager.LayoutParams.MATCH_PARENT;
        window.setAttributes(s);
        // endregion

        imgBack.setOnClickListener(this);
        lnOtherCloud.setOnClickListener(this);
        lnOtherLibrary.setOnClickListener(this);
        lnOtherCamera.setOnClickListener(this);
    }

    private void loadDownloadedFiles() {
        ArrayList<DownloadedFiles> files = getDownloadedFiles();
        if (files.size() > 0) {
            lvInApp.setLayoutManager(new LinearLayoutManager(mainAct));
            lvInApp.setHasFixedSize(true);
            lvInApp.setItemViewCacheSize(20);

            DownloadedFilesAdapter adapter = new DownloadedFilesAdapter(mainAct, files, this);
            lvInApp.setAdapter(adapter);

            tvNodata.setVisibility(View.GONE);
            lvInApp.setVisibility(View.VISIBLE);
        } else {
            tvNodata.setVisibility(View.VISIBLE);
            lvInApp.setVisibility(View.GONE);
        }
    }

    private File createImageFile() throws IOException {
        String imageFileName = "BPMON_" + System.currentTimeMillis() + ".png";
        File dirs = new File(mainAct.getFilesDir().toString(), "Camera");
        if (!dirs.exists()) {
            dirs.mkdirs();
        }
        return new File(dirs, imageFileName);
    }

    private void otherClould() {
        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
        intent.putExtra(Intent.EXTRA_MIME_TYPES, Constants.mimeTypes);
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
            intent.setType("*/*");
        } else {
            intent.setType("file/*");
        }

        if (mLauncherNormal != null) {
            mLauncherNormal.launch(intent);
        } else {
            mainAct.startActivityForResult(intent, requestCodeFile);
        }

        dialog.dismiss();
    }

    private void otherLibrary() {
        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
        intent.putExtra(Intent.EXTRA_MIME_TYPES, Constants.mimeTypesImage);

        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
            intent.setType("*/*");
        } else {
            intent.setType("file/*");
        }

        if (mLauncherNormal != null) {
            mLauncherNormal.launch(intent);
        } else {
            mainAct.startActivityForResult(intent, requestCodeFile);
        }

        dialog.dismiss();
    }

    private void otherCamera() {
        BaseActivity.sBaseActivity.checkUserPermission(new BaseActivity.PermissionListener() {
            @Override
            public void OnAcceptedAllPermission() {
                File file = null;
                Uri uri = null;
                try {
                    file = createImageFile();
                    file.createNewFile();

                    uri = FileProvider.getUriForFile(mainAct, mainAct.getPackageName() + ".provider", file);

                    switch (flagview) {
                        case Vars.FlagView.DetailWorkflow_Comment:
                        case Vars.FlagView.DetailWorkflow_ControlInputAttachmentVertical: {
                            ((DetailWorkflowActivity) mainAct).fileCamera = uri;
                            break;
                        }
                        case Vars.FlagView.ReplyComment: {
                            ((ReplyCommentFragment) baseFragment).fileImage = uri;
                            break;
                        }
                        case Vars.FlagView.DetailCreateTask_ControlInputAttachmentVertical:
                        case Vars.FlagView.DetailCreateTask_Comment: {
                            ((DetailCreateTaskActivity) mainAct).uri = uri;
                            break;
                        }
                        case Vars.FlagView.DetailCreateTask_Child_ControlInputAttachmentVertical: {
                            ((CreateChildTaskActivity) mainAct).uri = uri;
                            break;
                        }
                    }

                    Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
                    intent.putExtra(MediaStore.EXTRA_OUTPUT, uri);
                    if (mLauncherNormal != null) {
                        mLauncherCamera.launch(intent);
                    } else {
                        mainAct.startActivityForResult(intent, requestCodeCamera);
                    }
                    dialog.dismiss();
                } catch (Exception ex) {
                    dialog.dismiss();
                    Log.d("ERR Capture", ex.getMessage());
                }
            }

            @Override
            public void OnCancelPermission() {
            }

            @Override
            public void OnNeverRequestPermission() {

            }
        }, new String[]{
                Manifest.permission.CAMERA,
                Manifest.permission.WRITE_EXTERNAL_STORAGE,
                Manifest.permission.READ_EXTERNAL_STORAGE});
    }

    private ArrayList<DownloadedFiles> getDownloadedFiles() {
        ArrayList<DownloadedFiles> items = new ArrayList<>();
        File directory = new File(mainAct.getFilesDir().toString(), CurrentUser.getInstance().getUser().getID());
        File[] files = directory.listFiles();
        if (files != null) {
            for (int i = 0; i < files.length; i++) {
                DownloadedFiles file = new DownloadedFiles();
                file.setID("");
                file.setTitle(files[i].getName());
                file.setSize(files[i].length());
                file.setCreated(files[i].lastModified());
                file.setModified(files[i].lastModified());
                file.setPath(files[i].getPath());
                items.add(file);
            }
        }

        return items;
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.img_PopupControlAttachmentChooseFile_Back: {
                dialog.dismiss();
                break;
            }
            case R.id.ln_PopupControlAttachmentChooseFile_Other_Cloud: {
                otherClould();
                break;
            }
            case R.id.ln_PopupControlAttachmentChooseFile_Other_Library: {
                otherLibrary();
                break;
            }
            case R.id.ln_PopupControlAttachmentChooseFile_Other_Camera: {
                otherCamera();
                break;
            }
        }
    }

    @Override
    public void OnItemClick(DownloadedFiles file) {
        Intent intent = new Intent();
        intent.setAction(VarsReceiver.SELECT_FILE_INAPP);
        String type = "";

        if (requestCodeFile == Constants.mDetailAttachment || requestCodeFile == Constants.mTaskAttachment) {
            type = "inputattachmenthorizon";
        } else if (requestCodeFile == Constants.mDetailComment || requestCodeFile == Constants.mTaskComment) {
            type = "comment";
        }

        intent.putExtra("type",type);
        intent.putExtra("file", new Gson().toJson(file));
        BroadcastUtility.send(mainAct, intent);
        dialog.dismiss();
    }
}
